compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -115,7 +116,12 @@ fn resolve_associated_item<'tcx>(
     );
 
     let trait_ref = ty::TraitRef::from_method(tcx, trait_id, rcvr_substs);
-    let vtbl = tcx.codegen_fulfill_obligation((param_env, ty::Binder::bind(trait_ref, tcx)))?;
+    // FIXME: we should instead track bound variables from their origin,
```

Comment:
I really don't want someone to start using `BoundVarsCollector` for something else. As much as it *isn't* the right place for it, I think it's better to just move that here until this is fixed. (With a very clear comment on *why*)